dags/top-downloads-from-sowflake.py

```diff
@@ -0,0 +1,117 @@
+"""This script is used to execute a query on Snowflake and report the results to a
+slack channel. This retrieved the top X publicly downloaded Synapse projects.
+See ORCA-301 for more context."""
+
+from datetime import datetime
+
+from airflow.decorators import dag, task
+from airflow.models.param import Param
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+from airflow.models import Variable
+from slack_sdk import WebClient
+
+
+dag_params = {
+    "snowflake_conn_id": Param("SNOWFLAKE_SYSADMIN_PORTAL_RAW_CONN", type="string"),
+}
+
+dag_config = {
+    "schedule_interval": "0 18 * * *",
+    "start_date": datetime(2024, 2, 20),
+    "catchup": False,
+    "default_args": {
+        "retries": 1,
+    },
+    "tags": ["snowflake"],
+    "params": dag_params,
+}
+
+QUERY = """
+WITH PUBLIC_PROJECTS AS (
+    SELECT
+        node_latest.project_id,
+        node_latest.name
+    FROM
+        synapse_data_warehouse.synapse.node_latest
+    WHERE
+        node_latest.is_public AND
+        node_latest.node_type = 'project'
+),
+DEDUP_FILEHANDLE AS (
+    SELECT DISTINCT
+        PUBLIC_PROJECTS.name,
+        filedownload.user_id,
+        filedownload.file_handle_id AS FD_FILE_HANDLE_ID,
+        filedownload.record_date,
+        filedownload.project_id,
+        file_latest.content_size
+    FROM
+        synapse_data_warehouse.synapse.filedownload
+    INNER JOIN
+        PUBLIC_PROJECTS
+    ON
+        filedownload.project_id = PUBLIC_PROJECTS.project_id
+    INNER JOIN
+        synapse_data_warehouse.synapse.file_latest
+    ON
+        filedownload.file_handle_id = file_latest.id
+    WHERE
+        filedownload.record_date = DATEADD(HOUR, -24, CURRENT_DATE)
+),
+
+DOWNLOAD_STAT AS (
+    SELECT
+        name,
+        project_id,
+        count(record_date) AS DOWNLOADS_PER_PROJECT,
+        count(DISTINCT user_id) AS NUMBER_OF_UNIQUE_USERS_DOWNLOADED,
+        count(DISTINCT FD_FILE_HANDLE_ID) AS NUMBER_OF_UNIQUE_FILES_DOWNLOADED
+    FROM
+        DEDUP_FILEHANDLE
+    GROUP BY
+        project_id, name
+)
+SELECT
+    'https://www.synapse.org/#!Synapse:syn' || cast(DOWNLOAD_STAT.project_id as varchar) as project,
+    DOWNLOAD_STAT.name,
+    DOWNLOAD_STAT.DOWNLOADS_PER_PROJECT
+FROM
+    DOWNLOAD_STAT
+ORDER BY
+    DOWNLOADS_PER_PROJECT DESC NULLS LAST
+LIMIT 10;
+"""
+
+
+@dag(**dag_config)
+def snowflake_top_downloads_to_slack():
+    @task
+    def get_top_downloads_from_snowflake(**context) -> str:
+        snow_hook = SnowflakeHook(context["params"]["snowflake_conn_id"])
+        ctx = snow_hook.get_conn()
+        cs = ctx.cursor()
+        cs.execute(QUERY)
+        top_downloaded_df = cs.fetch_pandas_all()
+
+        message = ":synapse: Top Downloaded Public Synapse Projects Yesterday!\n\n"
```

Comment:
Nit: did we want to include the fact that it's yesterday UTC time?

Reply:
Like:
```suggestion
        message = ":synapse: Top Downloaded Public Synapse Projects Yesterday (UTC)!\n\n"
```
?